compiler/src/iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.cpp

```diff
@@ -83,10 +83,27 @@ static MMAAttr chooseIntrinsicMMAAttr(TypeRange eTypes, TargetWgpAttr wgp) {
 
 static DataTiledMMAAttr
 chooseDataTiledMMAAttr(TypeRange eTypes, TargetAttr target,
-                       IREE::Encoding::EncodingAttr encoding) {
+                       IREE::Encoding::EncodingAttr encoding,
+                       GPUEncodingResolverAttr resolver) {
   if (!target) {
     return {};
   }
+  // First try finding a data-tiled MMA layout through the ukernel provider if
+  // one can be found in the config. The ukernel provider is only available in
+  // case ukernels are enabled, so we're sure we want to override the default
+  // logic.
+  DictionaryAttr config = resolver.getConfiguration();
+  IREE::Codegen::UKernelProviderInterface ukernelProvider =
+      getUKernelProviderFromTarget(config);
+  if (ukernelProvider) {
```

Comment:
How about moving it into the if condition? We can rename it `provider` because the scope is small.

```suggestion
if (IREE::Codegen::UKernelProviderInterface ukernelProvider =
      getUKernelProviderFromTarget(config))
```

Reply:
Yeah, good suggestion.